src/python/pants/option/global_options.py

```diff
@@ -435,6 +438,9 @@ def register_bootstrap_options(cls, register):
              advanced=True)
     register('--process-execution-use-local-cache', type=bool, default=True, advanced=True,
              help='Whether to keep process executions in a local cache persisted to disk.')
+    register('--process-execution-local-enable-nailgun', type=bool, default=DEFAULT_EXECUTION_OPTIONS.process_execution_local_enable_nailgun,
+             help='Whether or not to use nailgun to run the requests that are marked as nailgunnable.',
+             advanced=True)
```

Comment:
bootstrap options may not be on Subsystems currently.

Reply:
We don't actually _need_ the `CommandRunner` to be on `Core`; it could happily live on `Session`, in which case we can pass the options _after_ options parsing has happened. The only real benefit of it being on `Core` is we can in theory keep gRPC connections alive across builds when using pantsd, but I'm sure we could find a way to stash the `Channel`s on some cache on `Core` which we use when constructing a `remote::CommandRunner`, or something.

Reply:
Could we move this discussion to an issue? It feels in a different scope as this PR, and I'd kind of like to close comments ASAP, as there are so many of them.

Reply:
https://github.com/pantsbuild/pants/issues/8524